File: ruby/engine/test/StackLevelTooDeepMeasure/measure.rb
########################################################################################################################
#  OpenStudio(R), Copyright (c) Alliance for Sustainable Energy, LLC.
#  See also https://openstudio.net/license
########################################################################################################################

class StackLevelTooDeepMeasure < OpenStudio::Measure::ModelMeasure

  def name
    return "Stack Level Too Deep Measure"
  end

  def modeler_description
     return "The arguments method calls an infinitely recursive function"
  end

  def s(x)
    return x * s(x-1)
  end

  def arguments(model)
    s(10)
    args = OpenStudio::Measure::OSArgumentVector.new

    return args
  end

  def run(model, runner, user_arguments)
    super(model, runner, user_arguments)

    return true
  end

end

StackLevelTooDeepMeasure.new.registerWithApplication

File: python/engine/test/StackLevelTooDeepMeasure/measure.py
########################################################################################################################
#  OpenStudio(R), Copyright (c) Alliance for Sustainable Energy, LLC.
#  See also https://openstudio.net/license
########################################################################################################################

import typing

import openstudio

def s(x):
    return s(x)

class StackLevelTooDeepMeasure(openstudio.measure.ModelMeasure):
    def name(self):
        return "Stack Level Too Deep Measure"

    def modeler_description(self):
        return "The arguments method calls an infinitely recursive function"

    def arguments(self, model: typing.Optional[openstudio.model.Model] = None):
        s(10)
        args = openstudio.measure.OSArgumentVector()

        return args

    def run(
        self,
        model: openstudio.model.Model,
        runner: openstudio.measure.OSRunner,
        user_arguments: openstudio.measure.OSArgumentMap,
    ):
        """
        define what happens when the measure is run
        """
        super().run(model, runner, user_arguments)  # Do **NOT** remove this line

StackLevelTooDeepMeasure().registerWithApplication()
